src/types.rs

```diff
@@ -27,8 +27,8 @@ impl<'a> ::conv::TryFrom<&'a str> for Command {
         // Parse out name and params
         if let Some(command) = words.next() {
 
-            // Strip leading slash
-            let name: String = command[1..].into();
+            // Strip leading slash and bot name following the @ symbol
+            let name: String = try!(command[1..].split('@').next().ok_or(CommandParseError::NoCommand)).into();
```

Comment:
Actually it's not necessary for a bot in private mode, because the bot only sees messages directed at him.

But I'll simply parse the bot name and add it to the command. Then it can be validated later on.